src/ILogger/ApplicationInsightsLoggerProvider.cs

```diff
@@ -0,0 +1,129 @@
+﻿// -----------------------------------------------------------------------
+// <copyright file="ApplicationInsightsLoggerProvider.cs" company="Microsoft">
+// Copyright (c) Microsoft Corporation. 
+// All rights reserved.  2013
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Microsoft.Extensions.Logging.ApplicationInsights
+{
+    using System;
+    using System.Collections.Concurrent;
+    using System.Collections.Generic;
+    using System.Text;
+    using Microsoft.ApplicationInsights;
+    using Microsoft.ApplicationInsights.Extensibility;
+    using Microsoft.Extensions.Options;
+
+    /// <summary>
+    /// Application insights logger provider.
+    /// </summary>
+    /// <seealso cref="Microsoft.Extensions.Logging.ILoggerProvider" />
+    /// <seealso cref="Microsoft.Extensions.Logging.ISupportExternalScope" />
+    [ProviderAlias("ApplicationInsights")]
+    public class ApplicationInsightsLoggerProvider : ILoggerProvider, ISupportExternalScope
+    {
+        /// <summary>
+        /// The application insights logger options.
+        /// </summary>
+        private readonly ApplicationInsightsLoggerOptions applicationInsightsLoggerOptions;
+
+        /// <summary>
+        /// The telemetry client to be used to log messages to Application Insights.
+        /// </summary>
+        private readonly TelemetryClient telemetryClient;
+
+        /// <summary>
+        /// The collection of application insights loggers stored against categoryName.
+        /// categoryName -> <see cref="ApplicationInsightsLogger"/> map.
+        /// </summary>
+        private readonly ConcurrentDictionary<string, ApplicationInsightsLogger> applicationInsightsLoggers
+            = new ConcurrentDictionary<string, ApplicationInsightsLogger>();
+
+        /// <summary>
+        /// The external scope provider to allow setting scope data in messages.
+        /// </summary>
+        private IExternalScopeProvider externalScopeProvider;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ApplicationInsightsLoggerProvider"/> class.
+        /// </summary>
+        /// <param name="telemetryConfiguration">The telemetry configuration.</param>
+        /// <param name="applicationInsightsLoggerOptions">The application insights logger options.</param>
+        /// <exception cref="System.ArgumentNullException">
+        /// telemetryConfiguration
+        /// or
+        /// loggingFilter
+        /// or
+        /// applicationInsightsLoggerOptions.
+        /// </exception>
+        public ApplicationInsightsLoggerProvider(
+            TelemetryConfiguration telemetryConfiguration,
+            IOptions<ApplicationInsightsLoggerOptions> applicationInsightsLoggerOptions)
```

Comment:
The current pattern of using `IOptions` doesn't react to configuration changes. Not sure how important it is for this implementation however.

Reply:
The way AI is configured today there isn't really an option to keep shifting the pipeline. Hence I went with IOptions and not IOptionsMonitor. If this changes, we can look into using IOptionsMonitor.

Reply:
Still @cijothomas  to confirm in case I overlooked a case.

Reply:
Yes IOptions works fine for now. (we can enhance later, but not required for this PR)

Reply:
Resolving this for now then.